src/engine/Collision/CollisionGroupManager.ts

```diff
@@ -0,0 +1,52 @@
+import { CollisionGroup } from './CollisionGroup';
+
+/**
+ * Static class for managing collision groups in excalibur, there is a maximum of 32 collision groups possible in excalibur
+ */
+export class CollisionGroupManager {
+  // using bitmasking the maximum number of groups is 32, because that is the heighest 32bit integer that JS can present.
```

Comment:
JS supports `2^53 - 1` [max safe integer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER), while not quite 64-bit, it's higher than 32-bit, does this limit still apply to this code?

Reply:
That's a good question, let me do some digging and find the correct answer here. We could potentially support additional groups.

Reply:
@kamranayub After experimenting in the tests this does not work unfortunately.

So the answer is tricky here, while JS does indeed support `2^53-1` integers, it does not support bitwise operators to that length before they are mangled by IEEE representation. JS bitwise operators only work on 32 bit signed integers unfortunately.

![image](https://user-images.githubusercontent.com/612071/58597560-22816780-823e-11e9-93b4-8b3c9e5d48b3.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#Signed_32-bit_integers
https://stackoverflow.com/questions/2983206/bitwise-and-in-javascript-with-a-64-bit-integer

Reply:
In fact you can see the signed numbers 32 integer overflow

![image](https://user-images.githubusercontent.com/612071/58597856-7771ad80-823f-11e9-8cac-fff894221e49.png)

Reply:
Well now we know! Thanks 👍